lib/ts/controllers/s-popover.ts

```diff
@@ -275,6 +280,172 @@ namespace Stacks {
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+    export function showPopover(element: Element) {
+        const [isPopover, controller, _] = getPopover(element);
+
+        if (controller) {
+            controller.show();
+        } else if (isPopover) {
+            element.setAttribute("data-s-popover-auto-show", "true");
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Helper to manually hide an s-popover element via external JS
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+    export function hidePopover(element: Element) {
+        const [isPopover, controller, _, popover] = getPopover(element);
+
+        if (controller) {
+            controller.hide();
+        } else if (isPopover) {
+            element.removeAttribute("data-s-popover-auto-show");
+            if (popover) {
+                popover.classList.remove("is-visible");
+            }
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Options to use when attaching a popover via `Stacks.attachPopover`.
+     * @see Stacks.attachPopover
+     */
+    export interface PopoverOptions {
+        /**
+         * When true, the `click->s-popover#toggle` action will be attached to the controller element or reference element.
+         */
+        toggleOnClick?: boolean;
+        /**
+         * When set, `data-s-popover-placement` will be set to this value on the controller element.
+         */
+        placement?: string;
+
+        /**
+         * When true, the popover will appear immediately when the controller connects.
+         */
+        autoShow?: boolean;
+    }
+
+    /**
+     * Attaches a popover to an element and performs additional configuration.
+     * @param element the element that will receive the `data-controller="s-popover"` attribute.
+     * @param popover an element with the `.s-popover` class or HTML string containing a single element with the `.s-popover` class.
+     *                If the popover does not have a parent element, it will be inserted as a immediately after the reference element.
+     * @param options an optional collection of options to use when configuring the popover.
+     */
+    export function attachPopover(element: Element, popover: Element | string, options?: PopoverOptions)
+    {
+        const [isPopover, controller, referenceElement, existingPopover] = getPopover(element);
+
+        if (existingPopover) {
+            throw `element already has popover with id="${existingPopover.id}"`
+        }
+
+        if (!referenceElement) {
+            throw `element has invalid data-s-popover-reference-selector attribute`
+        }
+
+        if (typeof popover === 'string') {
+            const elements = document.createRange().createContextualFragment(popover).children;
```

Comment:
PR Note: we do this exact type of string to dom insert in `tooltip`. Potential for abuse/xss was already discussed then, so we should be good here.

No action needed.